### PR TITLE
[WB-9831] Rename `wandb local` to `wandb server`

### PR DESCRIPTION
**Pull request by jsbroks**
Fixes [WB-9831](https://wandb.atlassian.net/browse/WB-9831)

Description
-----------
Renames `wandb local` to `wandb server`

### CODE REVIEW

File: wandb/cli/cli.py [hunk 1]
```diff
@@ -1417,10 +1417,37 @@ def docker(
...
 )
 @display_error
-def local(ctx, port, env, daemon, upgrade, edge):
+def local(*args, **kwargs):
+    wandb.termwarn(
+        "`wandb local` has been replaced with `wandb server`."
+    )
+    wandb.termlog(
+        "Please use `wandb server` instead."
+    )
```
Comment (kptkin) [inline wandb/cli/cli.py:1429]: wait do we want to keep both? that's how I read @raubitsj comment and only in the future move to `server` i'm fine doing it now... the only issue is if someone have some scripts that use `wandb local` they might break

**Review by kptkin — APPROVED** [review]
Leaving it up to you how you want to transition!

File: wandb/cli/cli.py (hunk 1 above)
Comment (vanpelt) [reply wandb/cli/cli.py:1429]: Agreed with @kptkin here.  There's no reason to break the old command.  Let's print the warning but just call `wandb server`:

```
def local(ctx, *args, **kwargs):
    wandb.termwarn(
        "`wandb local` has been replaced with `wandb server`."
    )
    ctx.invoke(server, *args, **kwargs) 
```

**Review by vanpelt — CHANGES_REQUESTED** [review]

**Comment by vanpelt** [comment]
@jsbroks I just made this handle all the cases, let me know how much you hate it ;)

image

File: wandb/cli/cli.py [hunk 2]
```diff
@@ -1421,10 +1423,55 @@ def docker(
...
+)
+@click.option(
+    "--env", "-e", default=[], multiple=True, help="Env vars to pass to wandb/local"
+)
+@click.option(
+    "--daemon/--no-daemon", default=True, help="Run or don't run in daemon mode"
+)
+@click.option(
+    "--upgrade", is_flag=True, default=False, help="Upgrade to the most recent version"
+)
```
Comment (jsbroks) [inline wandb/cli/cli.py:1469]: @vanpelt I think we should hide this flag and add it back if we decided not to do `wandb server upgrade`

**Comment by jsbroks** [comment]
I would argue we should not have `wandb server` as an alias for `wandb server start`. My reasoning: It makes the tool harder to use correctly (since its hard to discover the subcommands), other tools follow a help menu convention for sub commands, and its easier to add this feature later then to remove it.

E.g Here is googles CLI
image

**Comment by vanpelt** [comment]
My argument is that a user should never need to learn or know about any sub commands.  They really only need to be able to start a server.  What are the critical sub commands that we want the user to be able to discover by typing `wandb server`?

**Comment by vanpelt** [comment]
@jsbroks I implemented your suggestions and added a `wandb server stop` to provide some justification for making a new group of commands anyway.  Let's get this into the next release.  CC @sephmard

File: wandb/docker/__init__.py [hunk 3]
```diff
@@ -203,9 +204,10 @@ def image_id(image_name: str) -> Optional[str]:
     if "@sha256:" in image_name:
         return image_name
     else:
-        return shell(
-            ["inspect", image_name, "--format", "{{index .RepoDigests 0}}"]
-        ) or image_id_from_registry(image_name)
+        digests = shell(["inspect", image_name, "--format", "{{.RepoDigests}}"])
+        if digests == "[]" or digests is None:
+            return image_id_from_registry(image_name)
+        return json.loads(digests)[0]
```
Comment (jsbroks) [inline wandb/docker/__init__.py:210]: @vanpelt This is breaking stuff? Can we remove it

**Pull request merged**